frontend/test/__support__/e2e/helpers/e2e-misc-helpers.js

```diff
@@ -133,28 +133,80 @@ export function visitQuestion(id) {
 }
 
 /**
- * Visit a dashboard and wait for its query to load.
+ * Visit a dashboard and wait for the related queries to load.
  *
- * NOTE: Avoid using this helper if you need to explicitly wait for
- * and assert on the individual dashcard queries.
- *
- * @param {number} id
+ * @param {number} dashboard_id
  */
-export function visitDashboard(id) {
-  cy.intercept("GET", `/api/dashboard/${id}`).as("getDashboard");
-  // The very last request when visiting dashboard always checks the collection it is in.
-  // That is - IF user has the permission to view that dashboard!
-  cy.intercept("GET", `/api/collection/*`).as("getParentCollection");
-
-  cy.visit(`/dashboard/${id}`);
-
-  // If users doesn't have permissions to even view the dashboard,
-  // the last request for them would be `getDashboard`.
-  cy.wait("@getDashboard").then(({ response: { statusCode } }) => {
-    canViewDashboard(statusCode) && cy.wait("@getParentCollection");
+export function visitDashboard(dashboard_id) {
+  // Some users will not have permissions for this request
+  cy.request({
+    method: "GET",
+    url: `/api/dashboard/${dashboard_id}`,
+    // That's why we have to ignore failures
+    failOnStatusCode: false,
+  }).then(({ status, body: { ordered_cards } }) => {
+    const dashboardAlias = "getDashboard" + dashboard_id;
```

Comment:
Nitpick: `getDashboard ${dashboard_id}`

(if we aim for consistency with the following line)